Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -115,14 +116,25 @@ callStatusCallback :: EsqDBFlow m r => CallCallbackReq -> m CallCallbackRes
 callStatusCallback req = do
   let callStatusId = req.customField.callStatusId
   _ <- QCallStatus.findById callStatusId >>= fromMaybeM CallStatusDoesNotExist
-  runTransaction $ QCallStatus.updateCallStatus callStatusId (exotelStatusToInterfaceStatus req.status) req.conversationDuration req.recordingUrl
+  runTransaction $ QCallStatus.updateCallStatus callStatusId (exotelStatusToInterfaceStatus req.status) req.conversationDuration (Just req.recordingUrl)
   return Ack
 
-directCallStatusCallback :: EsqDBFlow m r => Text -> ExotelCallStatus -> Text -> Maybe Int -> m CallCallbackRes
+directCallStatusCallback :: EsqDBFlow m r => Text -> ExotelCallStatus -> Maybe Text -> Maybe Int -> m CallCallbackRes
 directCallStatusCallback callSid dialCallStatus recordingUrl_ callDuration = do
   callStatus <- QCallStatus.findByCallSid callSid >>= fromMaybeM CallStatusDoesNotExist
-  recordingUrl <- parseBaseUrl recordingUrl_
-  runTransaction $ QCallStatus.updateCallStatus callStatus.id (exotelStatusToInterfaceStatus dialCallStatus) (fromMaybe 0 callDuration) recordingUrl
+  let newCallStatus = exotelStatusToInterfaceStatus dialCallStatus
+  mbrecordingUrl <- case recordingUrl_ of
+    Just recordUrl -> do
+      if recordUrl == ""
+        then throwError CallStatusDoesNotExist
```

Comment:
How does recordUrl decoding related to existance of CallStatus in db? Why this exception?

Reply:
Why do you need to check for empty string? Even if recordUrl is empty string, you should not throw the error, consider that as Nothing and update the other values of CallStatus. Why to ignore other values just because recordUrl is empty.

Reply:
@RitikaHanish suggested to add check for empty string, exotel should either send null or url but not empty string. shall i remove this check?

Reply:
Changes done, resolving